src/cursor/cur_layered.c

```diff
@@ -666,6 +673,78 @@ __clayered_reset(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __clayered_copy_constituent_bound --
+ *     Copy the top level bound into a single constituent cursor
+ */
+static int
+__clayered_copy_constituent_bound(WT_CURSOR_LAYERED *clayered, WT_CURSOR *constituent)
+{
+    WT_CURSOR *base_cursor;
+    WT_SESSION_IMPL *session;
+
+    session = CUR2S(clayered);
+    base_cursor = (WT_CURSOR *)clayered;
+
+    if (constituent == NULL)
+        return (0);
+
+    /* Note that the inclusive flag is additive to upper/lower, so no need to check it as well */
+    if (F_ISSET(base_cursor, WT_CURSTD_BOUND_UPPER))
+        WT_RET(__wt_buf_dup(session, &constituent->upper_bound, &base_cursor->upper_bound));
+    else {
+        __wt_buf_free(session, &constituent->upper_bound);
+        WT_CLEAR(constituent->upper_bound);
+    }
+    if (F_ISSET(base_cursor, WT_CURSTD_BOUND_LOWER))
+        WT_RET(__wt_buf_dup(session, &constituent->lower_bound, &base_cursor->lower_bound));
+    else {
+        __wt_buf_free(session, &constituent->lower_bound);
+        WT_CLEAR(constituent->lower_bound);
+    }
+    /* Copy across all the bound configurations */
+    F_SET(constituent, F_MASK(base_cursor, WT_CURSTD_BOUND_ALL));
+    return (0);
+}
+
+/*
+ * __clayered_copy_bounds --
+ *     A method for copying (or clearing) bounds on constituent cursors within a layered cursor
+ */
+static int
+__clayered_copy_bounds(WT_CURSOR_LAYERED *clayered)
+{
+    WT_RET(__clayered_copy_constituent_bound(clayered, clayered->ingest_cursor));
+    WT_RET(__clayered_copy_constituent_bound(clayered, clayered->stable_cursor));
+    return (0);
+}
+
+/*
+ * __clayered_bound --
+ *     WT_CURSOR->bound method for the layered cursor type.
+ */
+static int
+__clayered_bound(WT_CURSOR *cursor, const char *config)
+{
+    WT_COLLATOR *collator;
+    WT_CURSOR_LAYERED *clayered;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    clayered = (WT_CURSOR_LAYERED *)cursor;
+
+    CURSOR_API_CALL(cursor, session, ret, bound, clayered->dhandle);
+    __clayered_get_collator(clayered, &collator);
+    /* Setup bounds on this top level cursor */
+    WT_ERR(__wti_cursor_bound(cursor, config, collator));
+    /* Copy those bounds into the constituents */
+    WT_ERR(__clayered_copy_bounds(clayered));
+
+err:
+    __clayered_leave(clayered);
```

Comment:
Is this deliberately missing a `__clayered_enter` call? If so, a comment (like what `__clayered_reset` has) would be good.